proxylib/proxylib/policymap.go

```diff
@@ -182,10 +182,6 @@ func newPortNetworkPolicies(config []*cilium.PortNetworkPolicy, dir string) Port
 			ParseError(fmt.Sprintf("Duplicate port number %d in (rule: %v)", port, rule), config)
 		}
 
-		if rule.GetProtocol() != core.SocketAddress_TCP {
-			ParseError(fmt.Sprintf("Invalid transport protocol %v", rule.GetProtocol()), config)
-		}
```

Comment:
The behavior is changed here: we used to panic if the protocol was unknown, now we simply skip it.

Reply:
Yes, I kept failing the connectivity test if I didn't do this. But good note. I'll call this out in my commit message.

Reply:
Before, it used to explicitly ignore UDP. Would it be better to also explicitly ignore SCTP as well?

Reply:
I think that would be best, yes, to preserve the previous behavior.

Reply:
Done.

Reply:
Reverted this change. For some reason I can no longer reproduce the failure. Maybe it will show up in the presubmits 🤷

Reply:
Furthermore, There is no way to explicitly skip SCTP because it simply doesn't exist as an option in the enum, so it's either no change at all or ignore all non-TCP protocols.